resources/io/jenkins/infra/docker/README.adoc

```diff
@@ -0,0 +1,130 @@
+
+= Jenkins Infra's Docker Tooling
+
+This page describes the generic instructions to build and verify the Docker images
+used by the Jenkins Infrastructure project.
+
+By "generic", we mean to have the same process and tooling reused for each image to provide:
+
+- Easier contribution and onboarding
+- Build reproductibility between contributor machines and CI environment
+- Best practises applied, including conventions, security patterns, standarized metadatas and optimized images
+- Try to leverage build security as much as possible
+
+== Quick Start
+
+* Ensure that you have the correct <<Requirements>> installed
+** If you have Docker installed and ready, please look at the section <<Using Docker>> first
+
+* Clone the repository hosting the source code of the Docker Image to build and position your command line inside this repository
++
+[source,bash]
+----
+git clone <URL of the repository docker-something>
+cd ./docker-something/
+----
+
+* Retrieve the `Makefile` located in the same directory as this documentation somewhere on your machine.
+It can be stored in your repository but you should git-ignore it.
+** Example using the `curl` command line:
++
+[source,bash]
+----
+# Download the Makefile. You might need to adapt the URL.
+curl -L "https://raw.githubusercontent.com/jenkins-infra/pipeline-library/master/resources/io/jenkins/infra/docker/Makefile" -O
+# Ensure that Makefile is git-ignored
+grep 'Makefile' .gitignore || echo 'Makefile' | tee -a .gitignore
+----
+
+* Execute the "developer" workflow with the command `make all`
+** Or `CONTAINER_BIN=docker make all` if you have Docker
+** And/or `make -C <Path to the directory containing your Makefile> all`
+
+
+== Requirements
+
+* Command prompt with bash (or zsh/ksh)
+
+* GNU Make 3.81+ command line
+
+* A https://github.com/moby/buildkit[Buildkit] compliant container tool
+** The command line top use is defined by the value of the environment variable `CONTAINER_BIN` which defauls to `img`
+** Our pipelines are using (Jessie Frazelle's) https://github.com/genuinetools/img[`genuinetools/img`] command line in v0.5.11, for security purposes (container in containers)
+** You can use https://docs.docker.com/get-docker/[Docker v20.10+] and its command line instead, by setting `CONTAINER_BIN` to `docker`
+
+* https://github.com/GoogleContainerTools/container-structure-test[Google's Container Structure Test] command line in v1.9.1+
```

Comment:
If this PR is only about https://github.com/jenkins-infra/docker-builder will it make sense to point to its README.md instead?

Reply:
In fact this PR (and the associated shared library) are targeted at any repository defining a Docker image to be built, in the organization `jenkins-infra` (at least?).

So https://github.com/jenkins-infra/docker-builder is the first one to really utilize the test part, however it is expected that other repos/pipelines are going to (ref. https://github.com/jenkins-infra/charts/blob/master/config/default/jenkins-infra.yaml#L178 => This is the JCasc config of the instance infra.ci.jenkins.io, which defines a "Docker Builds" GH org  folder). For instance: https://github.com/jenkins-infra/docker-datadog/, https://github.com/jenkins-infra/docker-terraform/, etc.

The trick here is that `docker-builder` is used by all, including itself: we are in a egg-and-chicken situation which makes the initial README confusing.
=> I confirm that this README is aimed to be generic for each of this image.